src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_errors.py

```diff
@@ -0,0 +1,125 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+# pylint: disable=line-too-long
+
+
+class ErrorClass:
+    error_title = ""
+    error_message = ""
+
+    def __init__(self, title, message):
+        self.error_title = title
+        self.error_message = message
+
+    def get_error_message(self, additional_message=None):
+        if additional_message:
+            return "An error occurred: {}\n{}\n{}".format(self.error_title, self.error_message, additional_message)
+        else:
+            return "An error occurred: {}\n{}".format(self.error_title, self.error_message)
+
+    def set_error_message(self, message):
+        return ErrorClass(self.error_title, message)
+
+    def append_error_message(self, message):
+        return self.set_error_message(self.error_message + "\n" + message)
+
+    def format_error_message(self, *args):
+        return self.set_error_message(self.error_message.format(*args))
+
+
+# DOCKER ERRORS
+
+
+DOCKER_COMMAND_ERROR = ErrorClass(
+    "DOCKER_COMMAND_ERROR",
+    "Please verify if Docker client is installed and running."
+)
+
+
+DOCKER_DAEMON_ERROR = ErrorClass(
+    "DOCKER_DAEMON_ERROR",
+    "Please verify if Docker daemon is running, or try restarting it."
+)
+
+
+DOCKER_VERSION_ERROR = ErrorClass(
+    "DOCKER_VERSION_ERROR",
+    "An error occured while retrieving Docker version. Please try verifying it manually."
+)
+
+
+DOCKER_PULL_ERROR = ErrorClass(
+    "DOCKER_PULL_ERROR",
+    "An error occurred while pulling a sample image. Please validate if all necessary components to pull an image are running properly."
```

Comment:
Can we be more specific about what components are necessary?

Reply:
I agree. Which should be the components? I can only think of docker daemon right now...

Reply:
Using network connection and docker daemon by now :)

Reply:
Yes. I will suggest network connection and docker daemon checks, it should be good by now.